python/ray/autoscaler/_private/_azure/config.py

```diff
@@ -79,26 +76,51 @@ def _configure_resource_group(config):
     with open(template_path, "r") as template_fp:
         template = json.load(template_fp)
 
-    # choose a random subnet, skipping most common value of 0
-    random.seed(resource_group)
-    subnet_mask = "10.{}.0.0/16".format(random.randint(1, 254))
+    # set unique id for resources in this cluster
+    unique_id = config["provider"].get("unique_id")
+    if unique_id is None:
+        hasher = sha256()
+        hasher.update(config["provider"]["resource_group"].encode("utf-8"))
+        hasher.update(config["cluster_name"].encode("utf-8"))
+        unique_id = hasher.hexdigest()[:UNIQUE_ID_LEN]
```

Comment:
What's the use case for `unique_id`? I imagine it's primarily if you want to change the cluster name without losing its identity? In that case, should we log the id?

Reply:
this is just to be able to isolate resources in azure based on the cluster, so they can live in the same resource group and not conflict. i did not want to rely on the cluster name itself because there are restrictions on characters and size of resource names, so i just hash it to create something mostly unique.

Reply:
Hmmm I see. If we want to just use it as an implementation detail, should we remove the cluster yaml entirely? 

If we want to treat it as an implementation detail, we probably shouldn't include it in the cluster yaml API at all. If we do want to expose it, I think we should probably INFO log as a matter of principle (logging the default values we fill in for explicitely exposed fields in general).

I defer to you on whether unique_id should be an implementation detail or a documented part of the node provider.

Reply:
i think it makes sense to log it, let me know if the latest update is what you had in mind